easybuild/easyconfigs/p/PyTorch/PyTorch-1.12.1_fix-skip-decorators.patch

```diff
@@ -0,0 +1,66 @@
+The decorators are implemented to run when the function is called which is after
+the test `setup` method spawned subprocesses which may use NCCL to sync failing when there are
+not enough GPUs available.
+So replace the custom code by calls to the `unittest` skip decorators.
+See https://github.com/pytorch/pytorch/issues/89686
+
+Author: Alexander Grund (TU Dresden)
+
+diff --git a/torch/testing/_internal/common_distributed.py b/torch/testing/_internal/common_distributed.py
+index 8baf7d03d9f..d4c17c80c5b 100644
+--- a/torch/testing/_internal/common_distributed.py
++++ b/torch/testing/_internal/common_distributed.py
+@@ -124,46 +124,19 @@ def skip_if_odd_worldsize(func):
+     return wrapper
+ 
+ def require_n_gpus_for_nccl_backend(n, backend):
+-    def decorator(func):
+-        @wraps(func)
+-        def wrapper(*args, **kwargs):
+-            if backend == "nccl" and torch.cuda.device_count() < n:
+-                sys.exit(TEST_SKIPS[f"multi-gpu-{n}"].exit_code)
+-            else:
+-                return func(*args, **kwargs)
+-
+-        return wrapper
+-
+-    return decorator
+-
++    return unittest.skipIf(backend == "nccl" and torch.cuda.device_count() < n,
++                           TEST_SKIPS[f"multi-gpu-{n}"].message)
+ 
+ def skip_if_lt_x_gpu(x):
+-    def decorator(func):
+-        @wraps(func)
+-        def wrapper(*args, **kwargs):
+-            if torch.cuda.is_available() and torch.cuda.device_count() >= x:
+-                return func(*args, **kwargs)
+-            sys.exit(TEST_SKIPS[f"multi-gpu-{x}"].exit_code)
+-
+-        return wrapper
+-
+-    return decorator
++    return unittest.skipUnless(torch.cuda.is_available() and torch.cuda.device_count() >= x,
++                               TEST_SKIPS[f"multi-gpu-{n}"].message)
+ 
+ 
+ # This decorator helps avoiding initializing cuda while testing other backends
+ def nccl_skip_if_lt_x_gpu(backend, x):
+-    def decorator(func):
+-        @wraps(func)
+-        def wrapper(*args, **kwargs):
+-            if backend != "nccl":
+-                return func(*args, **kwargs)
+-            if torch.cuda.is_available() and torch.cuda.device_count() >= x:
+-                return func(*args, **kwargs)
+-            sys.exit(TEST_SKIPS[f"multi-gpu-{x}"].exit_code)
+-
+-        return wrapper
+-
+-    return decorator
++    return unittest.skipUnless(backend != "nccl" or
++                               (torch.cuda.is_available() and torch.cuda.device_count() >= x),
++                               TEST_SKIPS[f"multi-gpu-{n}"].message)
```

Comment:
`TEST_SKIPS[f"multi-gpu-{n}"].message)` should be `TEST_SKIPS[f"multi-gpu-{x}"].message)`